Tests/VideoTests.swift

```diff
@@ -1,100 +1,211 @@
-                       "VideoManager.sample should return false for the viewing key")
+        let videoManager = VideoManager(trackerInstance: parselyTestTracker)
+
+        videoManager.trackPlay(
+            url: testUrl,
+            urlref: testUrl,
+            vId: testVideoId,
+            duration: TimeInterval(10),
+            metadata: nil,
+            extra_data: nil,
+            idsite: Parsely.testAPIKey
+        )
+        videoManager.trackPause()
+
+        expect(videoManager.sampleFn(key: self.testVideoKey)).to(beFalse())
     }
 
     func testHeartbeatFn() {
-        let testVideoKey: String = testUrl + "::" + testVideoId
-        let dummyEventArgs: Dictionary<String, Any> = videoManager!.generateEventArgs(
-            url: testUrl, urlref: "", extra_data: nil, idsite: Parsely.testAPIKey)
-        let dummyAccumulator: Accumulator = Accumulator(key: testVideoKey, accumulatedTime: 0, totalTime: 0,
-                                                        firstSampleTime: Date(),
-                                                        lastSampleTime: Date(), lastPositiveSampleTime: Date(),
-                                                        heartbeatTimeout: 0, contentDuration: 0, isEngaged: false,
-                                                        eventArgs: dummyEventArgs)
-        videoManager!.trackPlay(url: testUrl, urlref: testUrl, vId: testVideoId, duration: TimeInterval(10),
-                                metadata: nil, extra_data: nil, idsite: Parsely.testAPIKey)
-        videoManager!.heartbeatFn(data: dummyAccumulator, enableHeartbeats: true)
-        XCTAssertEqual(parselyTestTracker.eventQueue.length(), 2,
-                       "A call to VideoManager should add two events to eventQueue")
+        let videoManager = VideoManager(trackerInstance: parselyTestTracker)
+        let dummyAccumulator = makeAccumulator(videoManager: videoManager, key: testVideoKey, url: testUrl)
+
+        videoManager.trackPlay(
+            url: testUrl,
+            urlref: testUrl,
+            vId: testVideoId,
+            duration: TimeInterval(10),
+            metadata: nil,
+            extra_data: nil,
+            idsite: Parsely.testAPIKey
+        )
+        videoManager.heartbeatFn(data: dummyAccumulator, enableHeartbeats: true)
+
+        expect(self.parselyTestTracker.eventQueue.list).to(haveCount(2))
+    }
+
+    func testHeartbeatDoesNotCrashIfNoVideoHasBeenTracked() {
+        let videoManager = VideoManager(trackerInstance: parselyTestTracker)
+        let dummyAccumulator = makeAccumulator(videoManager: videoManager, key: testVideoKey, url: testUrl)
+
+        // Send heart beat without tracking a play first...
+        videoManager.heartbeatFn(data: dummyAccumulator, enableHeartbeats: true)
+
+        // ...and verify that no crash has occurred by performing a simple assertion on the tracker state
+        expect(self.parselyTestTracker.eventQueue.list).to(beEmpty())
```

Comment:
I didn't write it in the comment (can add it if you think it would be useful) but of course if there's a crash this code won't run. The assertion is there just so that we can have _something_ in the test.